crates/uv/tests/python_find.rs

```diff
@@ -270,7 +270,67 @@ fn python_find_project() {
     success: true
     exit_code: 0
     ----- stdout -----
-    [PYTHON-3.11]
+    [PYTHON-3.10]
+
+    ----- stderr -----
+    "###);
+
+    // But a pin should take precedence
+    uv_snapshot!(context.filters(), context.python_pin().arg("3.12"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    Pinned `.python-version` to `3.12`
+
+    ----- stderr -----
+    "###);
+    uv_snapshot!(context.filters(), context.python_find(), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    [PYTHON-3.12]
+
+    ----- stderr -----
+    "###);
+
+    // And we should warn if it's not compatible with the project
+    uv_snapshot!(context.filters(), context.python_pin().arg("3.10"), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: The requested Python version `3.10` is incompatible with the project `requires-python` value of `>=3.11`.
```

Comment:
Oh, we won't even let you write this pin! But we do need to handle the case where you've written it previously or with another tool.